core/sequencer/beam.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// To exclude beam dependencies run go with --tag nobeam.
+// +build !nobeam
+
+package sequencer
+
+import (
+	"context"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/beamx"
+
+	spb "github.com/google/keytransparency/core/sequencer/sequencer_go_proto"
+	tpb "github.com/google/trillian"
+)
+
+// CreateRevisionWithBeam is an implementation of CreateRevFn using beam.
+func CreateRevisionWithBeam(ctx context.Context, directoryID string, rev int64,
+	metaProto *spb.MapMetadata, batchSize int32,
+	mw MapWriter, mr MapReader, lr LogReader) error {
+
+	p := beam.NewPipeline()
```

Comment:
A single pipeline is used in the streaming case as well.  See this [example](https://github.com/apache/beam/blob/master/sdks/go/examples/streaming_wordcap/wordcap.go)

I'm thinking of having this function return the pipeline so that it can be used by a variety of runners.